### ISSUE TITLE
Update source files when new versions of papers come out

### ISSUE THREAD

**Issue by bfirsh** [arxiv-vanity/arxiv-vanity#374]
We don't do a good job of updating to newer versions of papers. I am currently doing this manually when people request.

There is a mechanism for doing this from the bulk source files, but clearly this is not working properly.

**Comment by jilljenn** [comment]
BTW I'm happy if you can update https://arxiv.org/abs/1712.06748v2
The problem is, I first sent to Vanity the link https://arxiv.org/abs/1712.06748
So by default it took v1 instead of v2, and after that it was not possible to force the refresh.
It should take the latest version as default, are you sure that it currently does that?

**Comment by wetneb** [comment]
Same issue here, when I generated https://www.arxiv-vanity.com/papers/1411.3827/ it picked the first version instead of the latest one.

So it is not even a problem of picking up updates after the paper has been first converted - when rendering a fresh paper, the LaTeX sources from the latest version should be used (which should be much easier than following updates afterwards).

**Comment by akvadrako** [comment]
Shouldn't you be able to specify the version in the URL ?

Like https://www.arxiv-vanity.com/papers/1712.01826v5/

**Comment by bfirsh** [comment]
#516 adds a `/papers/<arxiv_id>/update-version/` URL to force a paper to update. This should be automatic, but this is now possible in case anybody wants to update their own paper. Thanks @ryota-mo! :)

A potential improvement to that pull request: Cloudflare caches everything on arXiv Vanity, so that endpoint will probably only work once, then it'll get cached forever. It needs some no cache headers.

**Comment by chenjisen** [comment]
> #516 adds a `/papers/<arxiv_id>/update-version/` URL to force a paper to update. This should be automatic, but this is now possible in case anybody wants to update their own paper. Thanks @ryota-mo! :)
> 
> A potential improvement to that pull request: Cloudflare caches everything on arXiv Vanity, so that endpoint will probably only work once, then it'll get cached forever. It needs some no cache headers.

I still cannot update a paper.
One example:
[https://www.arxiv-vanity.com/papers/1707.01476/](url)
[https://arxiv.org/abs/1707.01476](url)

I tried [https://www.arxiv-vanity.com/papers/1707.01476/update-version](url) but it still do not change.

**Comment by bfirsh** [comment]
Yeah -- unfortunately that pull request was broken and actually broke normal papers (!) so I removed it.

If anyone wants to work on this, this is still open. I think it should be automatic, and totally could be. When a paper is rendered, it could check to see if there's a new version and render that instead.
